<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Help - CLASHub 1.0</title>
    <link rel="icon" href="Index_php/CLASH_favicon.png" type="image/png">
    <link rel="stylesheet" href="css/index2.css">
</head>
<body class="more-info-page">
    <header>
        <a href="index.html">
            <img src="Index_php/CLASHub_logo.png" alt="CLASH Logo" style="max-width: 600px; height: auto;">
        </a>
        <hr class="gradient-line">
    </header>

    <main>
        <section>
            <h2>Frequently Asked Questions (FAQs)</h2>
            <ul class="faq-list">
                <li>
                    <button class="faq-question">
                        1. What CLASH data is currently available in CLASHub? 
                        <span class="toggle-icon">+</span>
                    </button>
                    <div class="faq-answer">
                        <p>
                            CLASHub hosts data from four species: <strong>Human, Mouse, Drosophila melanogaster</strong>, and <strong>Caenorhabditis elegans</strong>. Below is the summary of available datasets:
                        </p>
                        <h4>Human</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>ZSWIM8 Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>A549</td><td>—</td><td>6</td><td>6</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34738798, SRR34738799, SRR34738800, SRR34738801, SRR34738802, SRR34738803, SRR34738804, SRR34738805, SRR34738790, SRR34738791, SRR34738792, SRR34738793</td></tr>
                                <tr><td>D425</td><td>3</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34757946, SRR34757949, SRR34757950</td></tr>
                                <tr><td>ES2</td><td>—</td><td>3</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34757940, SRR34757941, SRR34757942, SRR34757943, SRR34757944, SRR34757945</td></tr>
                                <tr><td>HCT116</td><td>5</td><td>—</td><td>3</td><td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE164634" target="_blank">GSE164634</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov//bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR13415087, SRR13415088, SRR13415089, SRR13415090, SRR13415091, SRR34757939, SRR34757947, SRR34757948</td></tr>
                                <tr><td>HEK293T</td><td>8</td><td>—</td><td>—</td><td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE198250" target="_blank">GSE198250</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov//bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR18281055, SRR18281057, SRR18281067, SRR18281068, SRR34761041, SRR34761042, SRR34761043, SRR34761044</td></tr>
                                <tr><td>HepG2</td><td>3</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34783077, SRR34783079, SRR34783080</td></tr>
                                <tr><td>Huh-7.5</td><td>12</td><td>—</td><td>—</td><td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE73057" target="_blank">GSE73057</a></td><td>SRR2413175, SRR2413176, SRR2413177, SRR2413178, SRR2413179, SRR2413180, SRR2413181, SRR2413182, SRR2413183, SRR2413184, SRR2413185, SRR2413186</td></tr>
                                <tr><td>H1299</td><td>—</td><td>3</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34768260, SRR34768261, SRR34768262, SRR34768263, SRR34768274, SRR34768275</td></tr>
                                <tr><td>Liver tissue</td><td>1</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34783163</td></tr>
                                <tr><td>MB002</td><td>—</td><td>4</td><td>4</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34783070, SRR34783071, SRR34783072, SRR34783073, SRR34783074, SRR34783075, SRR34783076, SRR34783078</td></tr>
                                <tr><td>MDA-MB-231</td><td>—</td><td>6</td><td>6</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR30817646, SRR30817647, SRR30817648, SRR30817649, SRR30817650, SRR30817651, SRR34738794, SRR34738795, SRR34738796, SRR34738797, SRR34738806, SRR34738807</td></tr>
                                <tr><td>OVCAR8</td><td>—</td><td>3</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34768264, SRR34768265, SRR34768266, SRR34768267, SRR34768276, SRR34768277</td></tr>
                                <tr><td>TIVE-EX-LTC</td><td>3</td><td>—</td><td>—</td><td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE101978" target="_blank">GSE101978</a></td><td>SRR5876947, SRR5876948, SRR5876949</td></tr>
                                <tr><td>T98G</td><td>—</td><td>3</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34743309, SRR34743310, SRR34743311, SRR34743312, SRR34743317, SRR34743318</td></tr>
                                <tr><td>U87MG</td><td>—</td><td>3</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34743313, SRR34743314, SRR34743315, SRR34743316, SRR34743319, SRR34743320</td></tr>                                
                                <tr><td>501Mel</td><td>—</td><td>3</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34768268, SRR34768269, SRR34768270, SRR34768271, SRR34768272, SRR34768273</td></tr>
                            </tbody>
                        </table>
    
                        <h4>Mouse</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Zswim8 Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>HE2.1B</td><td>6</td><td>—</td><td>—</td><td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE124687" target="_blank">GSE124687</a>
                                    </td><td>SRR8395242, SRR8395243, SRR8395244, SRR8395245, SRR8395246, SRR8395247</td></tr>
                                <tr><td>MEF</td><td>—</td><td>2</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34793109, SRR34793110, SRR34793111, SRR34793112</td></tr>
                                <tr>
                                    <td>Striatal cell</td>
                                    <td>—</td>
                                    <td>4</td>
                                    <td>4</td>
                                    <td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1093144" target="_blank">PRJNA1093144</a></td>
                                    <td>SRR28497185, SRR28497186, SRR28497189, SRR28497190, SRR2849718, 6SRR28497197, SRR28497198, SRR28497199, SRR28497200</td>
                                  </tr>
                                <tr><td>3T12</td><td>3</td><td>—</td><td>—</td><td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE124687" target="_blank">GSE124687</a>
                                    </td><td>SRR8395248, SRR8395249, SRR8395250</td></tr>
                                <tr><td>Cortex tissue</td><td>16</td><td>—</td><td>—</td><td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE73058" target="_blank">GSE73058</a>
                                    </td><td>SRR2413277, SRR2413278, SRR2413279, SRR2413282, SRR2413284, SRR2413286, SRR2413288, SRR2413289, SRR2413290, SRR2413293, SRR2413297, SRR2413298, SRR2413299, SRR2413300, SRR2413301, SRR2413302</td></tr>
                                <tr><td>Heart tissue</td><td>2</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34793107, SRR34793108</td></tr>
                                <tr><td>Kidney tissue</td><td>2</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1166120" target="_blank">PRJNA1166120</a></td><td>SRR34793105, SRR34793106</td></tr>
                            </tbody>
                        </table>
    
                        <h4>Drosophila melanogaster</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Dora Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>S2 cells</td><td>—</td><td>3</td><td>3</td><td>
                                    <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA896239" target="_blank">PRJNA896239</a>
                                    </td><td>SRR22129325, SRR22129327, SRR22129328, SRR22129284, SRR22129287, SRR22129298</td></tr>
                            </tbody>
                        </table>
    
                        <h4>Caenorhabditis elegans</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Ebax Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>Embryo</td><td>—</td><td>4</td><td>4</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE303817" target="_blank">GSE303817</a></td><td>—</td></tr>
                                <tr><td>L3 stage</td><td>—</td><td>7</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE56180" target="_blank">GSE56180</a></td><td>SRR1207389, SRR1207390, SRR1207391, SRR1207392, SRR1207393, SRR1207394, SRR1207395</td></tr>
                                <tr><td>mid-L4 stage</td><td>—</td><td>5</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA328816" target="_blank">PRJNA328816</a></td><td>SRR3882724, SRR3882728, SRR3882949, SRR3882950, SRR3882951</td></tr>
                            </tbody>
                        </table>
                    </div>
                </li>
                <li>
                    <button class="faq-question">
                        2. What Gene Expression Profile data is available in CLASHub? 
                        <span class="toggle-icon">+</span>
                    </button>
                    <div class="faq-answer">
                        <p>
                            Gene Expression Profile from four species: <strong>Human, Mouse, Drosophila melanogaster</strong>, and <strong>Caenorhabditis elegans</strong>. Below is the summary of available datasets:
                        </p>
                        <h4>Human</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>ZSWIM8 Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>A549</td><td>7</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE263036" target="_blank">GSE263036</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE212057" target="_blank">GSE212057</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE199309" target="_blank">GSE199309</a>
                                  </td><td>SRR28535493, SRR28535494, SRR28535495, SRR21237863, SRR21237869, SRR21237879, SRR18462418</td></tr>
                                <tr><td>D425</td><td>5</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE151810" target="_blank">GSE151810</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE185024" target="_blank">GSE185024</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE123760" target="_blank">GSE123760</a>
                                  </td><td>SRR11924485, SRR11924486, SRR16119415, SRR16119416, SRR8315029</td></tr>
                                <tr><td>ES2</td><td>6</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE218794" target="_blank">GSE218794</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE245778" target="_blank">GSE245778</a>
                                  </td><td>SRR22410790, SRR22410791, SRR22410792, SRR26439462, SRR26439463, SRR26439464</td></tr>
                                <tr><td>HEK293T</td><td>7</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE231583" target="_blank">GSE231583</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196043" target="_blank">GSE196043</a>
                                  </td><td>SRR24421974, SRR24421975, SRR24421976, SRR18074813, SRR18074814, SRR18074815, SRR18074816</td></tr>
                                <tr><td>Hela</td><td>7</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE273634" target="_blank">GSE273634</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE218727" target="_blank">GSE218727</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE199309" target="_blank">GSE199309</a>
                                  </td><td>SRR30058518, SRR30058519, SRR30058520, SRR22407570, SRR22407571, SRR22407572, SRR18462415</td></tr>
                                <tr><td>HepG2</td><td>5</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE224980" target="_blank">GSE224980</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE264010" target="_blank">GSE264010</a>
                                  </td><td>SRR28685775, SRR28685776, SRR28685777, SRR23387178, SRR23387179</td></tr>
                                <tr><td>H1299</td><td>4</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE212057" target="_blank">GSE212057</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE199309" target="_blank">GSE199309</a>
                                  </td><td>SRR21237865, SRR21237873, SRR21237881, SRR18462412</td></tr>
                                <tr><td>K562</td><td>6</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE199309" target="_blank">GSE199309</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167869" target="_blank">GSE167869</a>
                                  </td><td>SRR18462409, SRR13800753, SRR13800754, SRR13800737, SRR13800738, SRR13800739</td></tr>
                                <tr><td>MB002</td><td>4</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE261568" target="_blank">GSE261568</a>
                                  </td><td>SRR28341540, SRR28341541, SRR28341542,SRR28341543</td></tr>
                                <tr><td>MCF7</td><td>7</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE195761" target="_blank">GSE195761</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE178905" target="_blank">GSE178905</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163791" target="_blank">GSE163791</a>
                                  </td><td>SRR17944548, SRR17944549, SRR14915857, SRR14915858, SRR13296901, SRR13296902, SRR13296903</td></tr>
                                <tr><td>MDA-MB-231</td><td>3</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE178532" target="_blank">GSE178532</a>
                                  </td><td>SRR14870088, SRR14870089, SRR14870090</td></tr>
                                <tr><td>OVCAR8</td><td>4</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE246325" target="_blank">GSE246325</a>
                                  </td><td>SRR26536798, SRR26536799, SRR26536802, SRR26536803</td></tr>
                                <tr><td>T98G</td><td>5</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE112241" target="_blank">GSE112241</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=PRJNA580150" target="_blank">PRJNA580150</a>
                                  </td><td>SRR10358029, SRR10358030, SRR10358031, SRR6881782, SRR6881783</td></tr>
                                <tr><td>U87MG</td><td>6</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE147626" target="_blank">GSE147626</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235568" target="_blank">GSE235568</a>
                                  </td><td>SRR11433766, SRR11433767, SRR11433768, SRR24991947, SRR24991948, SRR24991949</td></tr>
                                <tr><td>501Mel</td><td>7</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=PRJNA515302" target="_blank">PRJNA515302</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE104869" target="_blank">GSE104869</a>
                                  </td><td>SRR8473015, SRR8473019, SRR8473020, SRR6163777, SRR6163778, SRR6163779, SRR6163780</td></tr>
                            </tbody>
                        </table>
                
                        <h4>Mouse</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Zswim8 Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>Adrenal Gland</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273702, SRR5273670, SRR5273654, SRR5273686</td></tr>
                                <tr><td>Bone Marrow</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273648, SRR5273680, SRR5273664, SRR5273696</td></tr>
                                <tr><td>Brain</td><td>8</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273637, SRR5273639, SRR5273641, SRR5273673, SRR5273657, SRR5273635, SRR5273689, SRR5273705</td></tr>
                                <tr><td>Embryonic Stem Cell</td><td>2</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJEB27315" target="_blank">PRJEB27315</a></td><td>ERR2640636, ERR2640637</td></tr>
                                <tr><td>Forestomach</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273662, SRR5273694, SRR5273678, SRR5273646</td></tr>
                                <tr><td>Heart</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273651, SRR5273683, SRR5273667, SRR5273699</td></tr>
                                <tr><td>iNeuron</td><td>3</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJEB27315" target="_blank">PRJEB27315</a></td><td>ERR2640652, ERR2640653, ERR2640654</td></tr>
                                <tr><td>Kidney</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273655, SRR5273671, SRR5273703, SRR5273687</td></tr>
                                <tr><td>Large Intestine</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273676, SRR5273692, SRR5273660, SRR5273644</td></tr>
                                <tr><td>Liver</td><td>8</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273636, SRR5273656, SRR5273640, SRR5273634, SRR5273638, SRR5273672, SRR5273704, SRR5273688</td></tr>
                                <tr><td>Lung</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273668, SRR5273700, SRR5273684, SRR5273652</td></tr>
                                <tr><td>MEF</td><td>3</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=GSE239373" target="_blank">GSE239373</a></td><td>SRR25443485, SRR25443484, SRR25443483</td></tr>
                                <tr><td>Muscle</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273643, SRR5273691, SRR5273659, SRR5273675</td></tr>
                                <tr><td>Neural Precursor</td><td>2</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJEB27315" target="_blank">PRJEB27315</a></td><td>ERR2640640, ERR2640641</td></tr>
                                <tr><td>Ovary</td><td>2</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273665, SRR5273649</td></tr>
                                <tr><td>Placenta</td><td>6</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=GSE252281" target="_blank">GSE252281</a></td><td>SRR27386997, SRR27386999, SRR27386998, SRR27387000, SRR27387002, SRR27387001</td></tr>
                                <tr><td>Skin</td><td>5</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=GSE222026" target="_blank">GSE222026</a></td><td>SRR22952493, SRR22952494, SRR22952490, SRR22952491, SRR22952492</td></tr>
                                <tr><td>Small Intestine</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273661, SRR5273693, SRR5273677, SRR5273645</td></tr>
                                <tr><td>Spleen</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273653, SRR5273685, SRR5273669, SRR5273701</td></tr>
                                <tr><td>Stomach</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273647, SRR5273663, SRR5273679, SRR5273695</td></tr>
                                <tr><td>Striatal cell</td><td>—</td><td>4</td><td>4</td><td>—</td><td>CS1, CS2, C1, C2, OS1, OS2, O1, O2</td></tr>
                                <tr><td>Testis</td><td>2</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273681, SRR5273697</td></tr>
                                <tr><td>Thymus</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273650, SRR5273682, SRR5273698, SRR5273666</td></tr>
                                <tr><td>Vesicular Gland</td><td>3</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA375882" target="_blank">PRJNA375882</a></td><td>SRR5273658, SRR5273674, SRR5273690</td></tr>
                            </tbody>
                        </table>
                
                        <h4>Drosophila melanogaster</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Dora Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>S2 cells</td><td>5</td><td>4</td><td>3</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196837" target="_blank">GSE196837</a>, 
                                    <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA896239" target="_blank">PRJNA896239</a>
                                  </td><td>SRR18048483, SRR18048484, SRR18048425, SRR18048423, SRR18048424, SRR22129330, SRR22129281, SRR22129317, SRR22129316, SRR18048427, SRR18048468, SRR18048426</td></tr>
                                <tr><td>0–4 h Embryos</td><td>4</td><td>—</td><td>—</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196837" target="_blank">GSE196837</a>
                                  </td><td>SRR18048437, SRR18048436, SRR18048435, SRR18048446</td></tr>
                                <tr><td>8–12 h Embryos</td><td>6</td><td>—</td><td>4</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196837" target="_blank">GSE196837</a>
                                  </td><td>SRR18048461, SRR18048433, SRR18048512, SRR18048481, SRR18048482, SRR18048434, SRR18048499, SRR18048531, SRR18048442, SRR18048532</td></tr>
                                <tr><td>12–16 h Embryos</td><td>6</td><td>—</td><td>4</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196837" target="_blank">GSE196837</a>
                                  </td><td>SRR18048539, SRR18048525, SRR18048508, SRR18048459, SRR18048432, SRR18048465, SRR18048448, SRR18048497, SRR18048529, SRR18048516</td></tr>
                                <tr><td>16–20 h Embryos wild type</td><td>5</td><td>—</td><td>4</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE196837" target="_blank">GSE196837</a>
                                  </td><td>SRR18048421, SRR18048538, SRR18048479, SRR18048463, SRR18048527, SRR18048542, SRR18048443, SRR18048495, SRR18048501</td></tr>
                                  <tr>
                                    <td>Fly Non-targeting Control</td>
                                    <td>—</td>
                                    <td>3</td>
                                    <td>—</td>
                                    <td>
                                      <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA896239" target="_blank">PRJNA896239</a>
                                    </td>
                                    <td>SRR22129292, SRR22129294, SRR22129296</td>
                                  </tr>
                            </tbody>
                        </table>
                
                        <h4>Caenorhabditis elegans</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Ebax Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>Embryos</td><td>4</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA922944" target="_blank">PRJNA922944</a></td><td>SRR23049957, SRR23049959, SRR23049928, SRR23049954</td></tr>
                                <tr><td>L1</td><td>6</td><td>2</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE68588" target="_blank">GSE68588</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE262626" target="_blank">GSE262626</a>,  <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267368" target="_blank">GSE267368</a></td><td>SRR2010468, SRR2010469, SRR28479531, SRR28479532, SRR28479533, SRR28479534</td></tr>
                                <tr><td>L2</td><td>3</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE266398" target="_blank">GSE266398</a></td><td>SRR28868053, SRR28868054, SRR28868055</td></tr>
                                <tr><td>L3</td><td>3</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA684142" target="_blank">PRJNA684142</a></td><td>SRR13238604, SRR13238605, SRR13238606</td></tr>
                                <tr><td>L4</td><td>3</td><td>—</td><td>—</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA922944" target="_blank">PRJNA922944</a></td><td>SRR23049963, SRR23049955, SRR23049961</td></tr>
                                <tr><td>Adult</td><td>4</td><td>1</td><td>1</td><td><a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA922944" target="_blank">PRJNA922944</a>,  <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267368" target="_blank">GSE267368</a></td><td>SRR23049965, SRR23049966, SRR23049906, SRR23049937</td></tr>
                            </tbody>
                        </table>
                    </div>
                </li>
                <li>
                    <button class="faq-question">
                        3. What miRNA Expression Profile data is available in CLASHub? 
                        <span class="toggle-icon">+</span>
                    </button>
                    <div class="faq-answer">
                        <p>
                            microRNA Expression Profile data from four species: <strong>Human, Mouse, Drosophila melanogaster</strong>, and <strong>Caenorhabditis elegans</strong>. Below is the summary of available datasets:
                        </p>

                        <h4>Human</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>Wild Type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>ZSWIM8 Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>A549</td><td>—</td><td>3</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163387" target="_blank">GSE163387</a></td><td>SRR13264637, SRR13264638, SRR13264639, SRR13264640, SRR13264641, SRR13264642</td></tr>
                                <tr><td>HEK293T</td><td>2</td><td>3</td><td>3</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE123627" target="_blank">GSE123627</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE158025" target="_blank">GSE158025</a>
                                  </td><td>SRR8311265, SRR8311266, SRR12650650, SRR12650651, SRR12650652, SRR12650653, SRR12650654, SRR12650655</td></tr>
                                <tr><td>HeLa</td><td>2</td><td>3</td><td>3</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE123627" target="_blank">GSE123627</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163387" target="_blank">GSE163387</a>
                                  </td><td>SRR13377179, SRR13377180, SRR13264643, SRR13264644, SRR13264645, SRR13264646, SRR13264647, SRR13264648</td></tr>
                                <tr><td>K562</td><td>6</td><td>—</td><td>6</td>  <td>
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE158025" target="_blank">GSE158025</a>,
                                    <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163388" target="_blank">GSE163388</a>
                                  </td><td>SRR12650656, SRR12650657, SRR12650658, SRR13264707, SRR13264708, SRR13264709, SRR12650659, SRR12650660, SRR12650661, SRR13264710, SRR13264711, SRR13264712</td></tr>
                                <tr><td>MCF7</td><td>—</td><td>2</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163388" target="_blank">GSE163388</a></td><td>SRR13264649, SRR13264650, SRR13264651, SRR13264652, SRR13264653</td></tr>
                            </tbody>
                        </table>

                        <h4>Mouse</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>wild type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Zswim8 Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>Brain</td><td>4</td><td>—</td><td>3</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235065" target="_blank">GSE235065</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148686" target="_blank">GSE148686</a></td><td>SRR24941005, SRR11547029, SRR24941026, SRR24940996, SRR24941021, SRR24941036, SRR24941000</td></tr>
                                <tr><td>Heart</td><td>4</td><td>—</td><td>5</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235065" target="_blank">GSE235065</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148686" target="_blank">GSE148686</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE231448" target="_blank">GSE231448</a></td><td>SRR24941003, SRR24941027, SRR24940995, SRR11547034, SRR24941022, SRR24391641, SRR24391629, SRR24941035, SRR24940999</td></tr>
                                <tr><td>Kidney</td><td>4</td><td>—</td><td>5</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235065" target="_blank">GSE235065</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148686" target="_blank">GSE148686</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE231448" target="_blank">GSE231448</a></td><td>SRR24941001, SRR24940993, SRR24941029, SRR11547035, SRR24941011, SRR24391611, SRR24391615, SRR24941033, SRR24941017</td></tr>
                                <tr><td>Liver</td><td>4</td><td>—</td><td>5</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235065" target="_blank">GSE235065</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148686" target="_blank">GSE148686</a></td><td>SRR24941004, SRR24940989, SRR24941030, SRR11547036, SRR24941010, SRR24391636, SRR24391616, SRR24941032, SRR24941016</td></tr>
                                <tr><td>Lung</td><td>4</td><td>—</td><td>5</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235065" target="_blank">GSE235065</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148686" target="_blank">GSE148686</a></td><td>SRR24940992, SRR24940998, SRR24941018, SRR24391640, SRR24941008, SRR24391628, SRR11547037, SRR24941031, SRR24941015</td></tr>
                                <tr><td>Intestine</td><td>3</td><td>—</td><td>5</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235065" target="_blank">GSE235065</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE231448" target="_blank">GSE231448</a></td><td>SRR24941002, SRR24940994, SRR24941028, SRR24941023, SRR24941012, SRR24391637, SRR24391617, SRR24941034</td></tr>
                                <tr><td>Neuron</td><td>—</td><td>3</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163387" target="_blank">GSE163387</a></td><td>SRR13264632, SRR13264633, SRR13264634, SRR13264635, SRR13264636</td></tr>
                                <tr><td>MEF</td><td>—</td><td>6</td><td>6</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163387" target="_blank">GSE163387</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE158025" target="_blank">GSE158025</a></td><td>SRR13264626, SRR13264627, SRR13264628, SRR12650662, SRR12650663, SRR12650664, SRR13264629, SRR13264630, SRR13264631, SRR12650665, SRR12650666, SRR12650667</td></tr>
                                <tr><td>Stomach</td><td>4</td><td>—</td><td>5</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235065" target="_blank">GSE235065</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148686" target="_blank">GSE148686</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE231448" target="_blank">GSE231448</a></td><td>SRR24941020, SRR24941009, SRR24940990, SRR11547042, SRR24941006, SRR24391639, SRR24941025, SRR24391619, SRR24941013</td></tr>
                                <tr><td>Skin</td><td>3</td><td>—</td><td>5</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE235065" target="_blank">GSE235065</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE231448" target="_blank">GSE231448</a></td><td>SRR24941019, SRR24940991, SRR24940997, SRR24941024, SRR24391638, SRR24941007, SRR24391618, SRR24941014</td></tr>
                                <tr><td>Striatal cell heterozygous</td><td>—</td><td>2</td><td>2</td><td>—</td><td>Cscr1, Cscr2, Cko1, Cko2</td></tr>
                                <tr><td>Striatal cell homozygous</td><td>—</td><td>2</td><td>2</td><td>—</td><td>HomoScr1, HomoScr2, HomoKo1, HomoKo2</td></tr>
                            </tbody>
                        </table>

                        <h4>Drosophila melanogaster</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>Wild Type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Dora Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>SRR Number</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>S2 cells</td>
                                    <td>3</td>
                                    <td>—</td>
                                    <td>3</td>
                                    <td>
                                      <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163388" target="_blank">GSE163388</a>
                                    </td>
                                    <td>SRR13264713, SRR13264714, SRR13264715, SRR13264716, SRR13264717, SRR13264718</td>
                                  </tr>
                            </tbody>
                        </table>

                        <h4>Caenorhabditis elegans</h4>
                        <table class="dataset-table">
                            <thead>
                                <tr>
                                    <th>Sample Name</th>
                                    <th>Wild Type (#)</th>
                                    <th>Non-targeting sgRNA Control (#)</th>
                                    <th>Ebax Knockout (#)</th>
                                    <th>BioProject Number</th>
                                    <th>Data Source</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr><td>Early Embryo</td><td>2</td><td>—</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267367" target="_blank">GSE267367</a></td><td>SRR29013903, SRR29013904, SRR29013905, SRR29013906</td></tr>
                                <tr><td>Late Embryo</td><td>2</td><td>—</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267367" target="_blank">GSE267367</a></td><td>SRR29013899, SRR29013900, SRR29013901, SRR29013902</td></tr>
                                <tr><td>L1</td><td>2</td><td>—</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267367" target="_blank">GSE267367</a></td><td>SRR29013895, SRR29013896, SRR29013897, SRR29013898</td></tr>
                                <tr><td>L2</td><td>2</td><td>—</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267367" target="_blank">GSE267367</a></td><td>SRR29013891, SRR29013892, SRR29013893, SRR29013894</td></tr>
                                <tr><td>L3</td><td>2</td><td>—</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267367" target="_blank">GSE267367</a></td><td>SRR29013887, SRR29013888, SRR29013889, SRR29013890</td></tr>
                                <tr><td>L4</td><td>3</td><td>—</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267367" target="_blank">GSE267367</a></td><td>SRR29013866, SRR29013867, SRR29013868, SRR29013869, SRR29013870</td></tr>
                                <tr><td>Gravid adult</td><td>2</td><td>—</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267367" target="_blank">GSE267367</a></td><td>SRR29013879, SRR29013880, SRR29013881, SRR29013882</td></tr>
                                <tr><td>Glp-4</td><td>2</td><td>—</td><td>2</td><td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE267367" target="_blank">GSE267367</a></td><td>SRR29013875, SRR29013876, SRR29013877, SRR29013878</td></tr>
                            </tbody>
                        </table>
                    </div>
                </li>
                <li>
                    <button class="faq-question">
                        4. How is CLASH data analyzed in CLASHub?
                        <span class="toggle-icon">+</span>
                    </button>
                    <div class="faq-answer">
                        <p>
                            <strong style="font-size: 1.2em;">Step 1: Data Upload and Input</strong><br>
                            CLASHub accepts paired-end <strong>FASTQ</strong> files or clean single-end <strong>FASTA</strong> files. Users need to provide minimal information to initiate the analysis.<br>
                            <strong>1.1 Paired-end Adapter Sequences:</strong><br>
                            5′ Adapter Sequence (default): <code>GATCGTCGGACTGTAGAACT</code><br>
                            3′ Adapter Sequence (default): <code>TGGAATTCTCGGGTGCCAAG</code><br>
                            <strong>1.2 Target species:</strong> (e.g., <em>Homo sapiens, Mus musculus, Drosophila melanogaster, Caenorhabditis elegans</em>)<br>
                            <strong>1.3 Output file names</strong><br>
                            <strong>1.4 Email address for result notification</strong>
                        </p>
                        
                        <p>
                            <strong style="font-size: 1.2em;">Step 2: Data Preprocessing</strong><br>
                            CLASHub automatically processes the uploaded data. For <strong>paired-end FASTQ</strong> files, the preprocessing pipeline includes:<br>
                            <strong>2.1 Adapter Trimming:</strong> Adapter sequences are removed using 
                            <code>
                                <a href="https://doi.org/10.14806/ej.17.1.200" target="_blank">cutadapt (v2.10) (Martin 2011)</a>
                            </code>.<br>

                            <strong>2.2 Read Merging:</strong> Overlapping paired-end reads are merged using 
                            <code>
                                <a href="https://doi.org/10.1093/bioinformatics/btt593" target="_blank">PEAR (v0.9.6) (Zhang et al. 2014)</a>
                            </code>.<br>
                            <strong>2.3 Redundancy Collapse:</strong> Redundant reads, often from PCR amplification, are collapsed using <code><a href="https://github.com/agordon/fastx_toolkit" target="_blank">fastx_collapser (v0.0.14)</a></code>.<br>
                            <strong>2.4 UMI Trimming:</strong> Unique Molecular Identifiers (UMIs) are trimmed to produce clean FASTA files.<br>
                            For <strong>single-end FASTA</strong> files, these preprocessing steps are not required.
                        </p>
                        
                        <p>
                            <strong style="font-size: 1.2em;">Step 3: Hybrid Identification</strong><br>
                            The cleaned data is processed to identify miRNA-target hybrids. The analysis uses:
                            <br><strong>3.1 <a href="https://doi.org/10.1016/j.ymeth.2013.10.015" target="_blank">hyb (Travis et al., 2014)</a>:</strong> Aligns reads to the reference transcript database using <code><a href="https://doi.org/10.1038/nmeth.1923" target="_blank">bowtie2 (v2.5.3)</a></code>.<br>
                            <strong>3.2 Reference Database:</strong> Includes 
                            <code>
                                <a href="https://doi.org/10.1093/nar/gkad1049" target="_blank">Ensembl genome (Harrison et al., 2024)</a>
                            </code> 
                            assemblies (e.g., GRCh38 for <em>Homo sapiens</em>, GRCm39 for <em>Mus musculus</em>, BDGP6.46 for <em>Drosophila melanogaster</em>, and WBcel235 for <em>Caenorhabditis elegans</em>) and mature miRNAs from 
                            <code>
                                <a href="https://doi.org/10.1093/nar/gky1141" target="_blank">miRBase (Release 22.1)</a>
                            </code>.
                            <br>
                            
                            <strong>3.3 Binding Stability:</strong> Free energy (<code>ΔG</code>, kcal/mol) is calculated using <code>UNAfold (v3.8)</code> (<a href="https://doi.org/10.1007/978-1-60327-429-6_1" target="_blank">Markham and Zuker, 2008</a>).
                        </p>
                        
                        <p>
                            <strong style="font-size: 1.2em;">Step 4: Conservation Score Calculation</strong><br>
                            Conservation scores are calculated using <code>CLASHub.py</code>, available on GitHub (<a href="https://github.com/UF-Xie-Lab/CLASHub/">https://github.com/UF-Xie-Lab/CLASHub/</a>), and phyloP tracks from the <strong>UCSC Genome Browser</strong>. These scores assess the evolutionary conservation of miRNA binding sites (<code>Perez et al. 2024</code>). The following phyloP tracks are used for different species:<br>
                            <strong>4.1</strong> <code>g38.phyloP100way</code> for <em>Homo sapiens</em> (human)<br>
                            <strong>4.2</strong> <code>mm39.phyloP35way</code> for <em>Mus musculus</em> (mouse)<br>
                            <strong>4.3</strong> <code>dm6.phyloP124way</code> for <em>Drosophila melanogaster</em> (fruit fly)<br>
                            <strong>4.4</strong> <code>ce11.phyloP135way</code> for <em>Caenorhabditis elegans</em> (nematode)
                        </p>
                        
                        <p>
                            <strong style="font-size: 1.2em;">Step 5: Hybrid Quantification and Site Type Analysis</strong><br>
                            All identified miRNA-target hybrids are quantified, and site types are classified based on sequence matching. The site types include: 
                            Offset 6mer, 6mer, 7mer-A1, 7mer-m8, and 8mer. These results are summarized into a comprehensive table, including expression levels and detailed hybrid information.
                        </p>
                        
                        <p>
                            <strong style="font-size: 1.2em;">Step 6: Output Results</strong><br>
                            The final output is presented in a detailed table and summary HTML report. The table includes the following columns:<br>
                            <strong>6.1</strong> miRNA Name (from the <code>miRBase</code> database)<br>
                            <strong>6.2</strong> Pairing Pattern (includes miRNA sequence, target sequence, and their base pairing relationships, analyzed via <code>UNAfold</code>)<br>
                            <strong>6.3</strong> Gene Name (from the <code>Ensembl</code> database)<br>
                            <strong>6.4</strong> Gene ID (from the <code>Ensembl</code> database)<br>
                            <strong>6.5</strong> Conservation Score (calculated using phyloP tracks from the UCSC Genome Browser: <code>g38.phyloP100way</code> for <em>Homo sapiens</em>, <code>mm39.phyloP35way</code> for <em>Mus musculus</em>, <code>dm6.phyloP124way</code> for <em>Drosophila melanogaster</em>, and <code>ce11.phyloP135way</code> for <em>Caenorhabditis elegans</em>)<br>
                            <strong>6.6</strong> Free Energy (ΔG, kcal/mol, calculated via <code>UNAfold</code>)<br>
                            <strong>6.7</strong> Gene Type (e.g., mRNA, lncRNA)<br>
                            <strong>6.8</strong> Element Region (e.g., CDS, 5'UTR, or 3'UTR)<br>
                            <strong>6.9</strong> Genomic Position<br>
                            <strong>6.10</strong> Binding Site Type (e.g., 8mer, 7mer-A1, 7mer-m8, 6mer, or non-seed match)<br>
                            <strong>6.11</strong> Number of Datasets with Hybrid Occurrence (specifies the datasets where hybrids occur, including wild-type samples, non-targeting sgRNA control samples, or ZSWIM8 knockout samples)<br>
                            <strong>6.12</strong> Normalized Hybrid Abundance (quantifies the abundance of hybrids across wild-type samples, non-targeting sgRNA control samples, or ZSWIM8 knockout samples)<br>
                        </p>
                    </div>
                </li>
                <li>
                    <button class="faq-question">
                        5. How is AQ-miRNA-seq data analyzed in CLASHub?
                        <span class="toggle-icon">+</span>
                    </button>
                    <div class="faq-answer">
                        <p>
                            <strong style="font-size: 1.2em;">Step 1: Data Upload and Input</strong><br>
                            Users upload miRNA sequencing data in one of three supported formats:<br>
                            <strong>1.1 Paired-End FASTQ (.gz):</strong> Requires 5′ and 3′ adapter sequences. Default adapter sequences:<br>
                            5′ Adapter Sequence (default): <code>GATCGTCGGACTGTAGAACT</code><br>
                            3′ Adapter Sequence (default): <code>TGGAATTCTCGGGTGCCAAG</code><br>
                            <strong>1.2 Single-End FASTQ (.gz):</strong> Requires only the 3′ adapter sequence. Default 3′ adapter:<br>
                            <strong>3′ Adapter Sequence (default):</strong> <code>TGGAATTCTCGGGTGCCAAG</code><br>
                            <strong>1.3 Cleaned Single-End FASTA (.gz):</strong> Does not require adapter sequences.<br>
                            Additional inputs include the output file name, target species (<em>Homo sapiens, Mus musculus, Drosophila melanogaster,</em> or <em>Caenorhabditis elegans</em>), and a valid email address for result notification.<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 2: Data Preprocessing</strong><br>
                            CLASHub processes uploaded data to produce clean FASTA files for analysis:<br>
                            <strong>2.1 Adapter Trimming:</strong> Adapter sequences are removed using 
                                <code>
                                    <a href="https://doi.org/10.14806/ej.17.1.200" target="_blank">cutadapt (v2.10) (Martin 2011)</a>
                                </code> 
                                for both paired-end and single-end FASTQ files.<br>

                            <strong>2.2 Read Merging:</strong> For paired-end FASTQ files, overlapping reads are merged using 
                            <code>
                                <a href="https://doi.org/10.1093/bioinformatics/btt593" target="_blank">PEAR (v0.9.6) (Zhang et al. 2014)</a>
                            </code>. 
                            Single-end files bypass this step.<br>
                            <strong>2.3 Redundancy Collapse:</strong> Redundant sequences, often from PCR amplification, are removed using <code>fastx_collapser (v0.0.14)</code>. This step applies to both paired-end and single-end files.<br>
                            <strong>2.4 UMI Trimming:</strong> Unique Molecular Identifiers (UMIs) are trimmed by removing four nucleotides from both 5′ and 3′ ends using <code>cutadapt</code>. This step applies to paired-end and single-end FASTQ files but is skipped for cleaned single-end FASTA files.<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 3: miRNA Identification and Quantification</strong><br>
                            The cleaned data is analyzed for miRNA quantification using a custom Python script, <code>CLASHub.py</code>, available on GitHub (<a href="https://github.com/UF-Xie-Lab/CLASHub/">https://github.com/UF-Xie-Lab/CLASHub/</a>).<br>
                            <strong>3.1 miRNA Mapping:</strong> Reads are aligned to mature miRNA sequences from <code><a href="https://doi.org/10.1093/nar/gky1141" target="_blank">miRBase (Release 22.1)</a></code>.<br>
                            <strong>3.2 Sequence Matching:</strong> The first 18 nucleotides of each read are matched to the corresponding miRNA in the database for accurate identification.<br>
                            <strong>3.3 Quantification:</strong> Total miRNA expression levels and isoform-specific abundances (capturing variations at the 3′ end) are calculated.<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 4: Output Results</strong><br>
                            The AQ-miRNA-seq analysis generates the following outputs:<br>
                            <strong>4.1 Total miRNA Table:</strong> Contains miRNA names and their total expression levels.<br>
                            <strong>4.2 Isoform Expression Table:</strong> Provides isoform-specific abundances, including sequence variations at the 3′ ends.<br>
                            <strong>4.3 Summary HTML Report:</strong> Includes key metrics such as the total number of uploaded reads, reads remaining after preprocessing, and the proportion of reads successfully aligned to known miRNAs.<br>
                        </p>
                    </div>
                </li>
                <li>
                    <button class="faq-question">
                        6. How is RNA-seq data analyzed in CLASHub?
                        <span class="toggle-icon">+</span>
                    </button>
                    <div class="faq-answer">
                        <p>
                            <strong style="font-size: 1.2em;">Step 1: Data Upload and Input</strong><br>
                            Users upload RNA sequencing data in paired-end <strong>FASTQ (.gz)</strong> format. Required inputs include:<br>
                            <strong>1.1 Adapter Sequences:</strong> Default adapter sequences for trimming are:<br>
                            5′ Adapter Sequence (default): <code>AGATCGGAAGAGCGTCGTGTA</code><br>
                            3′ Adapter Sequence (default): <code>AGATCGGAAGAGCACACGTCT</code><br>
                            <strong>1.2 Output File Name:</strong> User-defined name for the output files.<br>
                            <strong>1.3 Target Species:</strong> <em>Homo sapiens</em> (GRCh38), <em>Mus musculus</em> (GRCm39), <em>Drosophila melanogaster</em> (BDGP6), or <em>Caenorhabditis elegans</em> (WBcel235).<br>
                            <strong>1.4 Email Address:</strong> Results will be sent to the provided email.<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 2: Data Preprocessing</strong><br>
                            Uploaded data undergoes preprocessing to prepare for gene expression analysis:<br>
                            <strong>2.1 Adapter Trimming:</strong> Adapter sequences are removed using <code><a href="https://doi.org/10.14806/ej.17.1.200" target="_blank">Cutadapt (v2.10) (Martin 2011)</a></code>.<br>
                            <strong>2.2 Genome Mapping:</strong> Trimmed reads are aligned to the reference genome using <code><a href="https://doi.org/10.1038/nmeth.3317" target="_blank">HISAT2 (v2.2.1)</a></code>. The supported genome assemblies include:<br>
                            <strong>2.2.1</strong> <em>Homo sapiens</em> (GRCh38)<br>
                            <strong>2.2.2</strong> <em>Mus musculus</em> (GRCm39)<br>
                            <strong>2.2.3</strong> <em>Drosophila melanogaster</em> (BDGP6)<br>
                            <strong>2.2.4</strong> <em>Caenorhabditis elegans</em> (WBcel235)<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 3: Gene Expression Quantification</strong><br>
                            <strong>3.1 Gene Expression Quantification:</strong> Gene expression levels are quantified using <code><a href="https://doi.org/10.1038/nprot.2016.095" target="_blank">StringTie (v2.2.1)</a></code> to calculate Transcripts Per Million (TPM), providing normalized expression values for all detected genes.<br>
                        </p>
                
                        <p>
                            <strong>4. Differential Gene Expression Analysis:</strong> For identifying differentially expressed genes, raw counts are generated using the <code>prepDE.py3</code> script (part of StringTie). These raw counts are analyzed with <code><a href="https://doi.org/10.1186/s13059-014-0550-8" target="_blank">DESeq2 (v1.44)</a></code> to detect statistically significant changes in gene expression between experimental groups (e.g., treatment vs. control).<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 5: Output Results</strong><br>
                            The RNA-seq analysis generates the following outputs using a custom Python script, <code>CLASHub.py</code>, available on GitHub (<a href="https://github.com/UF-Xie-Lab/CLASHub/">https://github.com/UF-Xie-Lab/CLASHub/</a>):<br>
                            <strong>5.1 TPM Expression Table:</strong> Provides normalized gene expression values (TPM).<br>
                            <strong>5.2 Raw Count Table:</strong> Contains unprocessed gene-level counts for downstream analysis.<br>
                            <strong>5.3 DESeq2 Results Table:</strong> Includes log2 fold changes, adjusted p-values, and base mean values for differentially expressed genes.<br>
                            <strong>5.4 Summary HTML Report:</strong> Summarizes the total number of uploaded reads, trimmed reads, mapped reads, and key metrics for both gene expression quantification and differential analysis.<br>
                        </p>
                    </div>
                </li>
                <li>
                    <button class="faq-question">
                        7. How is cumulative fraction curve analysis performed in CLASHub?
                        <span class="toggle-icon">+</span>
                    </button>
                    <div class="faq-answer">
                        <p>
                            <strong style="font-size: 1.2em;">Step 1: Data Upload and Input</strong><br>
                            Users upload a differential gene expression CSV file with at least three required columns:<br>
                            <strong>1.1 GeneName:</strong> Specifies gene names used for analysis.<br>
                            <strong>1.2 BaseMean:</strong> Filters out genes with low expression (default threshold: 100).<br>
                            <strong>1.3 log2FoldChange:</strong> Indicates the direction and magnitude of differential regulation.<br>
                            Additional inputs include the target species (<em>Homo sapiens</em>, <em>Mus musculus</em>, <em>Drosophila melanogaster</em>, or <em>Caenorhabditis elegans</em>), the miRNA name, a BaseMean threshold, an output file name, and a valid email address for result delivery.<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 2: Target Identification</strong><br>
                            Target genes are classified into two groups for analysis:<br>
                            <strong>2.1 CLASH-Derived Targets:</strong><br>
                            • <strong>Conserved Targets:</strong> Identified in at least two cell types for <em>Homo sapiens</em> (15 cell types) and <em>Mus musculus</em> (7 cell types), containing 8mer or 7mer seed matches with conservation scores (phyloP > 0). For <em>Drosophila melanogaster</em>, conserved targets are restricted to the S2 cell line, and for <em>Caenorhabditis elegans</em>, they are limited to embryonic datasets.<br>
                            • <strong>All Targets:</strong> Includes all genes containing seed matches (8mer, 7mer, or 6mer), covering mRNA and ncRNA biotypes.<br>
                        </p>
                        
                        <p>
                            <strong>2.2 TargetScan-Derived Targets:</strong><br>
                            TargetScan predictions are derived from the file <code>Summary_Counts.txt</code>, which classifies targets into conserved and non-conserved categories:<br>
                            • <strong>Conserved Targets:</strong> miRNA targets specifically annotated as conserved in the TargetScan database.<br>
                            • <strong>All Targets:</strong> Includes both conserved and non-conserved sites, providing a comprehensive list of predicted miRNA interactions.<br>
                            Users can access the <code>Summary_Counts.txt</code> file for different species from the TargetScan database:<br>
                            • <em>Homo sapiens</em> (Human): Release 8, September 2021<br>
                            • <em>Mus musculus</em> (Mouse): Release 8, September 2021<br>
                            • <em>Drosophila melanogaster</em> (Fly): Release 7.2, October 2018<br>
                            • <em>Caenorhabditis elegans</em> (Worm): Release 6.2, June 2012<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 3: Cumulative Fraction Curve Generation</strong><br>
                            The uploaded gene expression data is used to compare fold change distributions between miRNA targets (identified via CLASH or TargetScan) and non-target genes. Genes with BaseMean values below the specified threshold are excluded to ensure robust results. The analysis generates cumulative fraction curves to visualize regulatory trends, such as miRNA upregulation leading to the repression of target genes relative to non-target genes.<br>
                        </p>
                
                        <p>
                            <strong style="font-size: 1.2em;">Step 4: Output Results</strong><br>
                            The cumulative fraction curve analysis produces the following outputs:<br>
                            <strong>4.1 Cumulative Fraction Curves:</strong> Visualize fold change distributions for miRNA targets versus non-targets.<br>
                            <strong>4.2 Summary Report:</strong> Includes key statistics such as the number of input genes, filtered genes (based on BaseMean), and target gene classifications (e.g., conserved or all targets).<br>
                            Results are summarized using a custom Python script, <code>CLASHub.py</code>, available on GitHub (<a href="https://github.com/UF-Xie-Lab/CLASHub/">https://github.com/UF-Xie-Lab/CLASHub/</a>).
                        </p>
                    </div>
                </li>
            </ul>
        </section>
    </main>

    <footer>
        <p>Copyright &copy; 2024 Xie Lab. Department of Biochemistry and Molecular Biology, University of Florida, USA.</p>
    </footer>
    <script src="more_info/faq_toggle.js"></script>
</body>
</html>